File: Data Structures & Algorithms/target-sum/submission-1.java
class Solution {
    public int findTargetSumWays(int[] nums, int target) {
        return findTargetSumWays(nums, 0, target);
    }
    private int findTargetSumWays(int[] nums, int index, int target) {
        if(index == nums.length) {
            return target == 0 ? 1 : 0;
        }
        return findTargetSumWays(nums, index + 1, target + nums[index]) + findTargetSumWays(nums, index + 1, target - nums[index]);
    }
}

File: Data Structures & Algorithms/combination-target-sum-ii/submission-1.py
import copy

class Solution:
    def combinationSum2(self, candidates: List[int], target: int) -> List[List[int]]:
        res = []
        sorted_candidates = sorted(candidates)
        self.comboSum2(sorted_candidates, 0, target, [], res)
        return res
    
    def comboSum2(self, candidates, idx, target, currCombo, res):
        if target == 0:
            res.append(copy.deepcopy(currCombo))
            return
        elif target < 0:
            return
        for i in range(idx, len(candidates)):
            candidate = candidates[i]
            if i > idx and candidate == candidates[i-1]:
                continue
            currCombo.append(candidate)
            self.comboSum2(candidates, i + 1, target - candidate, currCombo, res)
            currCombo.pop()